### PR TITLE
Add build-gh-pages presubmit job for node-feature-discovery

### PR DESCRIPTION
**Pull request by marquiz**
Verify buildability of gh-pages site.

### CODE REVIEW

File: config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-presubmits.yaml [hunk 1]
```diff
@@ -43,3 +43,22 @@ presubmits:
         - runner.sh
         args:
         - scripts/test-infra/build-image.sh
+  - name: pull-node-feature-discovery-build-gh-pages
+    always_run: true
+    decorate: true
+    annotations:
+      testgrid-dashboards: sig-node-node-feature-discovery
+      testgrid-tab-name: build-gh-pages
+      description: "Test build of gh-pages documentation"
```
Comment (fejta) [inline config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-presubmits.yaml:52]: Why does this require kind and/or a privileged security context?

**Review by fejta — APPROVED** [review]

File: config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-presubmits.yaml (hunk 1 above)
Comment (marquiz) [reply config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-presubmits.yaml:51]: Thanks for the review @fejta !

Kind is certainly not needed and was a copy-paste mistake. This is now fixed in the new version of the PR. Also dropped kind from the build-image job. Docs are built in docker, thus (AFAIU) dind and privileged context is needed. Please correct me if I'm wrong or if some other image than kubekins would be more appropriate.

**Review by fejta — APPROVED** [review]
/hold cancel

**Pull request merged**